File: chapter 1/ex_1_7.hs
module Ex1_7 where

average x y =
  (x + y) / 2

improve guess x =
  average guess (x / guess)

small_enough x =
  x / 100

good_enough guess x =
  abs (guess ** 2 - x) < small_enough x

sqrt_iter guess x =
  if good_enough guess x then print guess
  else sqrt_iter (improve guess x) x

File: chapter 1/ex_1_7.py
def average(x, y):
    return (x + y) / 2

def improve(guess, x):
    return average(guess, (x / guess))

def smallEnough(x):
    return x / 100

def goodEnough(guess, x):
    return abs(guess ** 2 - x) < smallEnough(x)

def sqrtIter(guess, x):
    return guess if goodEnough(guess, x) else sqrtIter(improve(guess, x), x)

print(sqrtIter(1, 9))
